util/BUILD

```diff
@@ -23,6 +23,18 @@ genrule(
     stamp = 1,  # this provides volatile-status.txt
 )
 
+genrule(
+    name = "scm_revision_file",
+    outs = ["scm_revision.txt"],
+    cmd = """awk '/BUILD_SCM_REVISION/ { print $$2 }' bazel-out/volatile-status.txt > $@""",
+    stamp = 1,  # this provides volatile-status.txt
+    tags = [
+        # Tell Bazel not to cache this target. Without the "no-cache" tag, this
+        # target happily produces a stale commit hash.
+        "no-cache",
```

Comment:
@cfrantz I noticed I was getting stale versions of `scm_revision.txt` when I rebuilt locally, but adding "no-cache" seemed to do the trick.

I'm surprised that `stamp = 1` wasn't enough — my mental model was that it would make `volatile-status.txt` a source of sorts, so this target should be rebuilt when `volatile-status.txt` changes.

Reply:
Actually, I think I fooled myself and I still am seeing stale commit hashes. I think that adding the tag convinced Bazel to rebuild scm_revision.txt, but it's not sufficient to rebuild again the next time that volatile-status.txt changes.

```sh
$ ./bazelisk.sh build-then 'git rev-parse HEAD ; cat %s' //util:scm_revision_file                      
[...]
226056a3aac2028ffdcb7420130a759359db6829
e20871e1cf102bf6492649eee4f8f7d3e59c9bb3
```

Reply:
Aha, I guess this is WAI.

>  In order to avoid rerunning stamped actions all the time though, Bazel pretends that the volatile file never changes.

https://bazel.build/docs/user-manual